<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Daniel_Osoko" />


<title>Report</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Report</h1>
<h4 class="author">Daniel_Osoko</h4>
<h4 class="date">23 3 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210325code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210325)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210325code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210325)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong5d21a86"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 5d21a86 </a>
</p>
</div>
<div id="strongRepositoryversionstrong5d21a86" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 5d21a86. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Report.Rmd</code>) and HTML (<code>docs/Report.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
5d21a86
</td>
<td>
Dosoko
</td>
<td>
2021-03-25
</td>
<td>
added a new analysis file
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="libraries" class="section level2">
<h2>Libraries</h2>
</div>
<div id="task-1" class="section level1">
<h1>Task 1</h1>
<div id="initial-step" class="section level2">
<h2>Initial Step</h2>
<p>To Begin the Analysis the Datasets have to be imported. In this case two datasets are used. The first dataset originates from small-seq and includes the counts of small RNAs in oocytes. The second dataset was obtained with the NEBNext protocol and contains the biotype counts of small RNAs in cauda epididymosomes. After importing both data sets all RNAs types are recalled to check for the RNAs of interest.</p>
<pre class="r"><code>b2 &lt;- read.csv(&quot;C:/Users/danio/Desktop/6th_Semester/Blockcourse/Study_of_Epigenetic_Mechanisms_in_Mental_Health/firstweek/b2.csv&quot;, row.names=1, header = TRUE, sep = &quot;,&quot;)
b2$type &lt;- as.factor(b2$type)


epy&lt;- read.csv(&quot;C:/Users/danio/Desktop/6th_Semester/Blockcourse/Study_of_Epigenetic_Mechanisms_in_Mental_Health/firstweek/BexceRpt_biotypeCounts.csv&quot;, row.names=1, header = TRUE, sep = &quot;,&quot;)</code></pre>
<p>Detected small RNAs in oocytes:</p>
<pre class="r"><code>levels(b2$type)</code></pre>
<pre><code> [1] &quot;IG_C_gene&quot;                          &quot;IG_C_pseudogene&quot;                   
 [3] &quot;IG_D_gene&quot;                          &quot;IG_D_pseudogene&quot;                   
 [5] &quot;IG_J_gene&quot;                          &quot;IG_LV_gene&quot;                        
 [7] &quot;IG_pseudogene&quot;                      &quot;IG_V_gene&quot;                         
 [9] &quot;IG_V_pseudogene&quot;                    &quot;lncRNA&quot;                            
[11] &quot;miRNA&quot;                              &quot;misc_RNA&quot;                          
[13] &quot;Mt_rRNA&quot;                            &quot;Mt_tRNA&quot;                           
[15] &quot;polymorphic_pseudogene&quot;             &quot;processed_pseudogene&quot;              
[17] &quot;protein_coding&quot;                     &quot;pseudogene&quot;                        
[19] &quot;ribozyme&quot;                           &quot;rRNA&quot;                              
[21] &quot;scaRNA&quot;                             &quot;scRNA&quot;                             
[23] &quot;snoRNA&quot;                             &quot;snRNA&quot;                             
[25] &quot;sRNA&quot;                               &quot;TEC&quot;                               
[27] &quot;TR_C_gene&quot;                          &quot;TR_D_gene&quot;                         
[29] &quot;TR_J_gene&quot;                          &quot;TR_J_pseudogene&quot;                   
[31] &quot;TR_V_gene&quot;                          &quot;TR_V_pseudogene&quot;                   
[33] &quot;transcribed_processed_pseudogene&quot;   &quot;transcribed_unitary_pseudogene&quot;    
[35] &quot;transcribed_unprocessed_pseudogene&quot; &quot;translated_unprocessed_pseudogene&quot; 
[37] &quot;unitary_pseudogene&quot;                 &quot;unprocessed_pseudogene&quot;            </code></pre>
<p>Detected small RNAs in cauda epididymosomes:</p>
<pre class="r"><code>rownames(epy)</code></pre>
<pre><code> [1] &quot;miRNA&quot;                              &quot;tRNA&quot;                              
 [3] &quot;protein_coding&quot;                     &quot;snRNA&quot;                             
 [5] &quot;piRNA&quot;                              &quot;misc_RNA&quot;                          
 [7] &quot;retained_intron&quot;                    &quot;snoRNA&quot;                            
 [9] &quot;Mt_rRNA&quot;                            &quot;processed_transcript&quot;              
[11] &quot;processed_pseudogene&quot;               &quot;nonsense_mediated_decay&quot;           
[13] &quot;TEC&quot;                                &quot;lincRNA&quot;                           
[15] &quot;Mt_tRNA&quot;                            &quot;antisense&quot;                         
[17] &quot;scaRNA&quot;                             &quot;ribozyme&quot;                          
[19] &quot;unprocessed_pseudogene&quot;             &quot;transcribed_processed_pseudogene&quot;  
[21] &quot;pseudogene&quot;                         &quot;rRNA&quot;                              
[23] &quot;sense_intronic&quot;                     &quot;circularRNA&quot;                       
[25] &quot;transcribed_unprocessed_pseudogene&quot; &quot;IG_C_gene&quot;                         
[27] &quot;translated_processed_pseudogene&quot;    &quot;IG_V_gene&quot;                         
[29] &quot;sense_overlapping&quot;                  &quot;bidirectional_promoter_lncRNA&quot;     
[31] &quot;transcribed_unitary_pseudogene&quot;     &quot;polymorphic_pseudogene&quot;            
[33] &quot;non_stop_decay&quot;                     &quot;IG_V_pseudogene&quot;                   
[35] &quot;unitary_pseudogene&quot;                 &quot;translated_unprocessed_pseudogene&quot; 
[37] &quot;IG_J_gene&quot;                          &quot;macro_lncRNA&quot;                      
[39] &quot;scRNA&quot;                              &quot;sRNA&quot;                              
[41] &quot;TR_C_gene&quot;                          &quot;TR_J_gene&quot;                         
[43] &quot;TR_V_gene&quot;                          &quot;TR_V_pseudogene&quot;                   
[45] &quot;exogenous_miRNA&quot;                    &quot;exogenous_rRNA&quot;                    
[47] &quot;exogenous_genomes&quot;                 </code></pre>
</div>
<div id="overview-of-data" class="section level2">
<h2>Overview of Data</h2>
<p>In the next steps the non-essential biotypes are combined into the group “others”, the biotypes of interest and therefore excluded from “others” are miRNA, snoRNA, rRNA and protein coding RNA .</p>
<p>The following table shows all small RNAs measurements from the oocyte analysis:</p>
<pre class="r"><code>levels(b2$type) [c(1:10,12,13:15,16,18,19,21,22,24:38)] &lt;- &quot;others&quot;
table(b2$type)</code></pre>
<pre><code>
        others          miRNA protein_coding           rRNA         snoRNA 
         29415           2207          21926            354           1507 </code></pre>
<p>In the case of the cauda epididymosomes the data was pre configurated, which explains the amount of entries per small RNA.</p>
<pre class="r"><code>epy$type &lt;- as.factor(row.names(epy))
levels(epy$type)[c(1:12, 14:22, 24:26, 28:31, 33:47)] &lt;- &quot;others&quot; 
table(epy$type)</code></pre>
<pre><code>
        others          miRNA protein_coding           rRNA         snoRNA 
            43              1              1              1              1 </code></pre>
<pre class="r"><code>library(reshape)</code></pre>
<pre><code>Warning: Paket &#39;reshape&#39; wurde unter R Version 4.0.4 erstellt</code></pre>
<pre><code>
Attache Paket: &#39;reshape&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    expand, rename</code></pre>
<pre class="r"><code>counts_b2&lt;- matrix(nrow=5, ncol= 53, dimnames = list(levels(as.factor(b2$type)), colnames(b2[,1:53]))) #creates empty table with column and row names from b2
          for (i in 1:53) { counts_b2[,i] &lt;- tapply( b2[,i], b2$type, FUN=sum)} #sums up values per category

counts_b2_normalized &lt;- prop.table(counts_b2, margin=2) #normalizes data margin 2=rows                                                        1=columns
    counts_b2_normalized &lt;- counts_b2_normalized*100 #turns values from 1 to 100%
    
counts_b2_normalized_Transposed &lt;- t(counts_b2_normalized) #needed for proper melt                                                                function
b2_norm_melt &lt;- melt(counts_b2_normalized_Transposed)
names(b2_norm_melt) &lt;- c(&quot;Cell&quot;, &quot;Type&quot;, &quot;Percentage&quot;)

#need melt for not normalized:

counts_b2_Transposed &lt;- t(counts_b2)
b2_raw_melt &lt;- melt(counts_b2_Transposed)
names(b2_raw_melt) &lt;- c(&quot;Cell&quot;, &quot;Type&quot;, &quot;Counts&quot;)


counts_epy&lt;- matrix(nrow=5, ncol= 10, dimnames = list(levels(as.factor(epy$type)), colnames(epy[,1:10]))) #creates empty table with column and row names from b2
          for (i in 1:10) { counts_epy[,i] &lt;- tapply( epy[,i], epy$type, FUN=sum)} #sums up values per category here only important for others

counts_epy_normalized &lt;- prop.table(counts_epy, margin=2) #normalizes data margin                                                        2=rows 1=columns
    counts_epy_normalized &lt;- counts_epy_normalized*100 #turns values from 1 to 100%
    
counts_epy_normalized_Transposed &lt;- t(counts_epy_normalized) #needed for proper melt                                                                function
epy_norm_melt &lt;- melt(counts_epy_normalized_Transposed)
names(epy_norm_melt) &lt;- c(&quot;Cell&quot;, &quot;Type&quot;, &quot;Percentage&quot;)

#need melt for not normalized:

counts_epy_Transposed &lt;- t(counts_epy)
epy_raw_melt &lt;- melt(counts_epy_Transposed)
names(epy_raw_melt) &lt;- c(&quot;Cell&quot;, &quot;Type&quot;, &quot;Counts&quot;)</code></pre>
</div>
<div id="visualisation-of-the-data" class="section level2">
<h2>Visualisation of the Data</h2>
<p>To visualize the importance of normalisation to avoid the PCR-bias (GC-content influences the duplication speed), the datasets are displayed in two graphs, with the first one displaying the ratios with raw read counts and the second graph displaying the same data after normalisation.</p>
<div id="oocytes" class="section level3">
<h3>Oocytes</h3>
<pre class="r"><code>library(ggplot2)

ggplot(data=b2_raw_melt,  aes(x=Cell, y=Counts, fill= Type)) +
  geom_bar(stat=&quot;identity&quot;) +
  scale_x_discrete(guide = guide_axis(angle = -90))+
  labs(title= &quot;Oocyte small RNA libraries&quot;, y=&quot;Counts&quot;, x = &quot;Cell&quot;)</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=b2_norm_melt,  aes(x=Cell, y=Percentage, fill= Type)) +
  geom_bar(stat=&quot;identity&quot;) +
  scale_x_discrete(guide = guide_axis(angle = -90))+
  labs(title= &quot;Normalised Oocyte small RNA libraries&quot;, y=&quot;Percentage&quot;, x = &quot;Cell&quot;)</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cauda-epididymosomes" class="section level3">
<h3>Cauda Epididymosomes</h3>
<pre class="r"><code>library(ggplot2)



ggplot(data=epy_raw_melt,  aes(x=Cell, y=Counts, fill= Type)) +
  geom_bar(stat=&quot;identity&quot;) +
  scale_x_discrete(guide = guide_axis(angle = -90))+
  labs(title= &quot;Cauda epididymosomes small RNA libraries&quot;, y=&quot;Counts&quot;, x = &quot;Cell&quot;)</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=epy_norm_melt,  aes(x=Cell, y=Percentage, fill= Type)) +
  geom_bar(stat=&quot;identity&quot;) +
  scale_x_discrete(guide = guide_axis(angle = -90))+
  labs(title= &quot;Normalised epididymosomes small RNA libraries&quot;, y=&quot;Percentage&quot;, x = &quot;Cell&quot;)</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>On a first glance it is visible that the cauda epididymosomes dont include any rRNA. Additionally by comparing the two normalised graphs one can deduce that the oocytes express more rRna and snoRNA than the cauda epigidymosomes but less miRNA.</p>
</div>
</div>
</div>
<div id="task-2" class="section level1">
<h1>Task 2</h1>
<div id="data" class="section level2">
<h2>Data</h2>
<pre class="r"><code>counts2 &lt;- read.csv(&quot;C:/Users/danio/Desktop/6th_Semester/Blockcourse/Study_of_Epigenetic_Mechanisms_in_Mental_Health/firstweek/counts2.csv&quot;, row.names=1, header = TRUE, sep = &quot;,&quot;)


lof &lt;- read.csv(&quot;C:/Users/danio/Desktop/6th_Semester/Blockcourse/Study_of_Epigenetic_Mechanisms_in_Mental_Health/firstweek/List_of_cells.csv&quot;, header = TRUE, sep = &quot;;&quot;)

counts_def &lt;- counts2[c(10,14:15),1:38] #removing types and non-Msus/CTRL samples
counts_lof &lt;- counts_def[,-1]
rownames(counts_lof) &lt;- counts_def[,1] #1st column to rownames

lof2 &lt;- lof[c(1:3,7:16,32:55),]</code></pre>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>As seen before, normalization is important to allow proper comparison between reads. The following graphs express the oocyte gene expressions in boxplots, the first one is the raw read, the second one express the normalized value, in which the bottom line is the same, giving us the foundation for proper comparison.</p>
<pre class="r"><code>#Creating/combining wanted data
dge &lt;- DGEList(counts=counts_lof, samples=lof2, group = lof2$Group) 

dge &lt;- calcNormFactors(dge) #TMM


#normalization
norm_dge &lt;- cpm(dge, log = TRUE)

boxplot(counts_lof, main = &quot;Boxplot of log counts&quot;, las=2) #before norm</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(norm_dge, main = &quot;Boxplot of normalized log counts&quot;, las=2) #after norm</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>To see the differences inbetween the oocytes the clustered are visiualized using PCA. The Cells are described by the two gene expressions which offer the biggest variances.</p>
<pre class="r"><code> #PCA after normalisation
#prcomp(lof, scale=FALSE)

df &lt;- t(norm_dge)
row.names(df) &lt;- paste(lof2$Group) #turns cell names (a1:a55) into treatment/groups

df_pca &lt;- prcomp(df) #use df as it is normalized and make pca where x describes de diffrent principal components
df_out &lt;- as.data.frame(df_pca$x)
df_out$group &lt;- sapply( strsplit(as.character(row.names(df)), &quot;_&quot;), &quot;[[&quot;, 1 ) #adds row names as column behind it as column &quot;group&quot;

percentage &lt;- round(df_pca$sdev / sum(df_pca$sdev) * 100, 2) #expresses counts as %
percentage &lt;- paste( colnames(df_out), &quot;(&quot;, paste( as.character(percentage), &quot;%&quot;, &quot;)&quot;, sep=&quot;&quot;) ) #describes the percentage each PC covers

group &lt;- lof2$Group

pca_CTRL_vs_MSUS &lt;- ggplot(df_out, aes(x=PC1, y=PC2, color=group)) #starting ggplot
pca_CTRL_vs_MSUS &lt;- pca_CTRL_vs_MSUS + #producing PCA
  geom_point(size = 2) + xlab(percentage[1]) + ylab(percentage[2]) +
  theme_classic() + theme(legend.title = element_blank()) +
  ggtitle(&quot;PCA plot CTRL and MSUS&quot;) + theme(plot.title = element_text(size = 20, hjust=0.5)) + 
  theme(
  axis.text.x = element_blank(),
  axis.text.y = element_blank(),
  axis.ticks = element_blank())

pca_CTRL_vs_MSUS</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="design-matrix" class="section level2">
<h2>Design matrix</h2>
<p>To properly compare the effects of Control and MSUS first a design matrix which expresses which sample belongs to which group in binary is needed.</p>
<pre class="r"><code>group &lt;- lof2$Group

design &lt;- model.matrix(~group)
design</code></pre>
<pre><code>   (Intercept) groupMSUS
1            1         0
2            1         0
3            1         0
4            1         1
5            1         1
6            1         1
7            1         1
8            1         1
9            1         1
10           1         1
11           1         1
12           1         1
13           1         0
14           1         0
15           1         0
16           1         0
17           1         1
18           1         1
19           1         1
20           1         1
21           1         1
22           1         1
23           1         1
24           1         1
25           1         1
26           1         0
27           1         0
28           1         0
29           1         0
30           1         0
31           1         0
32           1         0
33           1         0
34           1         0
35           1         0
36           1         0
37           1         0
attr(,&quot;assign&quot;)
[1] 0 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$group
[1] &quot;contr.treatment&quot;</code></pre>
</div>
<div id="differential-gene-expression-analysis" class="section level2">
<h2>Differential gene expression analysis</h2>
<p>The next step is to analyse which genes are differantly expressed. To accomplish this a linear function is fitted to the data and their expression is analysed statistically.</p>
<pre class="r"><code># Estimate dispersion
dge &lt;- estimateDisp(dge, design, robust=TRUE)

# Use voom transformation
v &lt;- voom(dge, design, plot=FALSE)

fit_limma &lt;- lmFit(v, design)
fit_limma &lt;- eBayes(fit_limma)

out_limma &lt;- topTable(fit_limma, adjust=&quot;BH&quot;, n = &quot;Inf&quot;)
head (out_limma)</code></pre>
<pre><code>                logFC  AveExpr         t      P.Value   adj.P.Val          B
CT010467.1 -0.5405604 17.94575 -3.605789 0.0008132907 0.002439872 -0.6924148
Gm24601     0.2149692 18.74298  3.140563 0.0030695317 0.004604297 -2.1084320
n.R5.8s1    0.1717594 18.06796  1.752960 0.0868361356 0.086836136 -4.8248902</code></pre>
</div>
<div id="volcano-plot" class="section level2">
<h2>Volcano plot</h2>
<pre class="r"><code>##Highlight genes that have an absolute fold change &gt; 2 and FDR &lt; 0.05
out_limma$threshold1 = as.factor(abs(out_limma$logFC) &gt; 2 &amp; out_limma$adj.P.Val &lt; 0.05)

cbPalette&lt;- c(&quot;mediumblue&quot;, &quot;firebrick1&quot;)

volcano_oocyte = ggplot(data=out_limma, aes(x=logFC, y=-log10(P.Value), colour=threshold1)) +
  geom_point(size=2) +
  xlab(&quot;log2 fold change&quot;) + ylab(&quot;-log10 p-value&quot;) + theme_classic() + theme(legend.position=&quot;none&quot;) + scale_color_manual (values=cbPalette) + 
  ggtitle(&quot;FDR  &lt; 0.05 &quot;) + theme(plot.title = element_text(hjust=0.5))

volcano_oocyte</code></pre>
<p><img src="figure/Report.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=German_Switzerland.1252  LC_CTYPE=German_Switzerland.1252   
[3] LC_MONETARY=German_Switzerland.1252 LC_NUMERIC=C                       
[5] LC_TIME=German_Switzerland.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] reshape_0.8.8               ggplot2_3.3.3              
 [3] DESeq2_1.30.0               SummarizedExperiment_1.20.0
 [5] Biobase_2.50.0              MatrixGenerics_1.2.0       
 [7] matrixStats_0.57.0          GenomicRanges_1.42.0       
 [9] GenomeInfoDb_1.26.0         IRanges_2.24.0             
[11] S4Vectors_0.28.0            BiocGenerics_0.36.0        
[13] edgeR_3.32.1                limma_3.46.0               
[15] workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] httr_1.4.2             bit64_4.0.5            splines_4.0.3         
 [4] blob_1.2.1             GenomeInfoDbData_1.2.4 yaml_2.2.1            
 [7] pillar_1.4.6           RSQLite_2.2.1          lattice_0.20-41       
[10] glue_1.4.2             digest_0.6.27          RColorBrewer_1.1-2    
[13] promises_1.1.1         XVector_0.30.0         colorspace_2.0-0      
[16] plyr_1.8.6             htmltools_0.5.0        httpuv_1.5.4          
[19] Matrix_1.2-18          XML_3.99-0.5           pkgconfig_2.0.3       
[22] genefilter_1.72.0      zlibbioc_1.36.0        purrr_0.3.4           
[25] xtable_1.8-4           scales_1.1.1           whisker_0.4           
[28] later_1.1.0.1          BiocParallel_1.24.1    git2r_0.28.0          
[31] tibble_3.0.4           annotate_1.68.0        farver_2.0.3          
[34] generics_0.1.0         ellipsis_0.3.1         withr_2.3.0           
[37] survival_3.2-7         magrittr_1.5           crayon_1.3.4          
[40] memoise_1.1.0          evaluate_0.14          fs_1.5.0              
[43] tools_4.0.3            lifecycle_0.2.0        stringr_1.4.0         
[46] munsell_0.5.0          locfit_1.5-9.4         DelayedArray_0.16.0   
[49] AnnotationDbi_1.52.0   compiler_4.0.3         rlang_0.4.8           
[52] grid_4.0.3             RCurl_1.98-1.2         rstudioapi_0.13       
[55] labeling_0.4.2         bitops_1.0-6           rmarkdown_2.5         
[58] gtable_0.3.0           DBI_1.1.0              R6_2.5.0              
[61] knitr_1.30             dplyr_1.0.2            bit_4.0.4             
[64] rprojroot_2.0.2        stringi_1.5.3          Rcpp_1.0.5            
[67] vctrs_0.3.4            geneplotter_1.68.0     tidyselect_1.1.0      
[70] xfun_0.19             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
